docs/adr/adr-005-data-retention.md

```diff
@@ -0,0 +1,201 @@
+
+Below are the decisions we made.
+
+### Max retention of 1 year for everything that's unused by the system
+
+There is no evidence of needing unused data for analytics beyond 1 year and so
+we can classify this old data as historic and remove it.
+
+We can split this unused data into two categories.
+
+Event data - which is only used for a short period after being created, and
+subsequently for analytics, which includes:
+
+* ContentChange & Message - removed once their `created_at` value is over a
+  year old.
+
+* MatchedContentChange & MatchedMessage - cascaded when their parent content
+  change / message is deleted.
+
+* DigestRun - removed once their `created_at` value is over a year old.
+
+* DigestRunSubscriber - cascaded when their parent digest run gets deleted.
+
+Subscriber data - which is used indefinitely until it's marked as "ended", which
+includes:
+
+* Subscription - removed when they've been ended for over a year, based on
+  their `ended_at` value. Once a subscription is ended it [can't currently be
+  made active again] and so can be deleted.
+
+* SubscriberList - removed when they have no active subscriptions and their
+  latest, ended subscription is over a year old.
+
+* Subscriber - removed once their `created_at` is over a year old and they
+  don't have any subscriptions (active or ended).
+
+### Shorter retention of 7 days for subscriber lists with no subscriptions
+
+We also remove subscriber lists which have no subscriptions (active or ended)
+and are over 7 days old. As mentioned in the context section, dead subscriber
+lists pollute our analytics and should be removed regularly.
+
+A scenario of when this can happen is when a user quits their signup journey
+midway through. 7 days was the chosen retention period to allow users to complete
+their signup journey as they have a [7 day limit] to confirm their
+subscription.
+
+### Modified the nullify process and retired the deactivated concept
+
+The [deactivated] concept was a point of confusion in the email system
+and wasn't actually needed so has been removed. It revolved around the
+`deactivated_at` field on the subscriber model. This field was mainly used as a
+means of determining when a subscriber could be nullified. We can however
+obtain this information via the most recent ended subscription of the
+subscriber (based on the subscriptions `ended_at`) which means `deactivated_at`
+is superfluous.
+
+In doing this we've also [modified the nullify process] which now only
+nullifies subscriber's addresses if they have no active subscriptions and their
+most recent is over 28 days old. We also nullify subscribers which have no
+subscriptions and are older than 28 days which prevents us from storing user's
+personal data longer than needed, if for whatever reason, a subscriber ends up
+with no subscriptions. See examples of where this happened [here][sub bug] and
+[here][sub bug two].
+
+### Not to modify the retention for emails (and subscription contents)
```

Comment:
Minor: we could move this section earlier, to help a future reader understand why we don't mention these models in previous sections.

Reply:
Sure, this is now the first section.